portalnet/src/overlay_service.rs

```diff
@@ -1651,6 +1708,10 @@ where
                     }
                 }
             }
+            // explicitally drop permit in new thread so it it is included in the move
```

Comment:
`so it it is included in the move` is also confusing. Typically, it's much easier to understand comments if you avoid vague nouns like `it`